purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -214,6 +217,37 @@ internal class PurchasesOrchestrator constructor(
 
     // region Public Methods
 
+    fun syncAttributesAndOfferingsIfNeeded(
+        callback: SyncAttributesAndOfferingsCallback,
+    ) {
+        val receiveOfferingsCallback = object : ReceiveOfferingsCallback {
+            override fun onReceived(offerings: Offerings) {
+                callback.onSuccess(offerings)
+            }
+
+            override fun onError(error: PurchasesError) {
+                callback.onError(error)
+            }
+        }
+
+        if (!lastSyncAttributesAndOfferingsRateLimiter.shouldProceed()) {
+            log(
+                LogIntent.WARNING,
+                SyncAttributesAndOfferingsStrings.RATE_LIMIT_REACHED.format(
+                    lastSyncAttributesAndOfferingsRateLimiter.maxCalls,
+                    lastSyncAttributesAndOfferingsRateLimiter.periodSeconds,
+                ),
+            )
+
+            getOfferings(receiveOfferingsCallback)
```

Comment:
Just to confirm, this will return the offerings from cache, not from network right?

Reply:
Correct correct!